internal/ibazel/fswatcher/fsevents/fsevents.go

```diff
@@ -96,6 +99,30 @@ func newEvent(name string, mask fsevents.EventFlags) (common.Event, bool) {
 	return e, true
 }
 
+// Find the longest common root path of all directories to watch.
+func findCommonRoot(names []string) []string {
```

Comment:
This feels like some rather complicated logic that could be trivially validated with a table test. Would you be willing to add one?

Reply:
The test could look something like this in `fsevents_test.go`

```go
// Copyright 2022 The Bazel Authors. All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

//go:build darwin
// +build darwin

package fsevents

import (
  "github.com/google/go-cmp/cmp"
)

func TestFindCommonRoot(t *testing.T) {
  tests := []struct{
    in []string
    want []string
  }{
    {
       []string{"a/b/c"},
       []string{"a/b/c"},
    },
    // Plus a few more test cases in here to demonstrate how it's supposed to work
  }
  for _, test := range tests {
    got := findCommonRoot(test.in)
    if diff := cmp.Diff(got, test.want); diff != "" {
      t.Errorf("findCommonRoot diff (-got,+want):\n%s", diff)
    }
  }
}
```

Reply:
Thanks!